README.md

```diff
@@ -1,2 +1,91 @@
 # queries
+
 Hosts reusable log queries which are built into a single queries.json file.
+
+## Query database structure
+
+```yaml
+queries:
+  pip-no-matching-distribution:  # also known as query "id", (sort key)
+    pattern: "ERROR: No matching distribution found for"  # required, can be list!
+    # optionals:
+    name: "PIP failed to find package"  # optional (aka 'reason'), id is used instead
+    files: []  # list of glob patterns, narrows down searching
+    regex: false  # optional, default is false
+    url: ... # str or list[str], issue links to lp, bz,...
+    category: ... # see Categories
+    tags: []  # see Tags
+    suppress-graph: false  # used for elastic-recheck
+    # uncommon fields
+    multiline: true  # https://opendev.org/openstack/ansible-role-collect-logs/src/branch/master/vars/sova-patterns.yml#L47
+  another-query-name: ...
+```
+
+Both [elastic-search](https://www.elastic.co/guide/en/elasticsearch/reference/current/term-level-queries.html) and [artcl](https://opendev.org/openstack/ansible-role-collect-logs) can make use of `regex` searches.
+
+Pattern is supposed to be an exact string match and if multiple are present
+we could easily convert them into a regex or logstash expression that uses
+logical `AND`.
+
+### Pattern
+
+On elastic-rechheck queries we have cases with multiple entries used on
+patterns, like `message:foo AND message:bar`. This is why we also allow
```

Comment:
I'd put option for logical NOT as well.
Like:
```yaml
pattern:
not_pattern:
```

Reply:
That is a good idea but I am afraid that we will need to embed the not inside the pattern text, like
```
pattern:  # this should build: message:foo AND NOT message:bar
 - foo
 - !bar
```

Still, as I do not have cases yet, I will defer it for the moment when we have first entry that needs it.